core/src/main/scala/better/files/ThreadBackedFileMonitor.scala

```diff
@@ -60,15 +57,12 @@ abstract class ThreadBackedFileMonitor(val root: File, maxDepth: Int) extends Fi
     }
   }
 
-  override def start() = {
+  override def start()(implicit executionContext: ExecutionContext) = {
     watch(root, maxDepth)
-    thread.start()
+    executionContext.execute(() => Iterator.continually(service.take()).foreach(process))
```

Comment:
> As there's only one WatchService, doesn't this mean there'll only be one thread active from the pool?

No it submits task to the execution context which figures out the best way to run this. This API let's you provide your ThreadPool or ExecutorService.